configs/_base_/datasets/dior.py

```diff
@@ -0,0 +1,77 @@
+# dataset settings
+dataset_type = 'DIORDataset'
+data_root = 'data/DIOR/'
+file_client_args = dict(backend='disk')
+
+train_pipeline = [
+    dict(type='mmdet.LoadImageFromFile', file_client_args=file_client_args),
+    dict(type='mmdet.LoadAnnotations', with_bbox=True, box_type='qbox'),
+    dict(type='ConvertBoxType', box_type_mapping=dict(gt_bboxes='rbox')),
+    dict(type='mmdet.Resize', scale=(800, 800), keep_ratio=True),
+    dict(
+        type='mmdet.RandomFlip',
+        prob=0.75,
+        direction=['horizontal', 'vertical', 'diagonal']),
+    dict(type='mmdet.PackDetInputs')
+]
+val_pipeline = [
+    dict(type='mmdet.LoadImageFromFile', file_client_args=file_client_args),
+    dict(type='mmdet.Resize', scale=(800, 800), keep_ratio=True),
+    # avoid bboxes being resized
+    dict(type='mmdet.LoadAnnotations', with_bbox=True, box_type='qbox'),
+    dict(type='ConvertBoxType', box_type_mapping=dict(gt_bboxes='rbox')),
+    dict(
+        type='mmdet.PackDetInputs',
+        meta_keys=('img_id', 'img_path', 'ori_shape', 'img_shape',
+                   'scale_factor'))
+]
+test_pipeline = [
+    dict(type='mmdet.LoadImageFromFile', file_client_args=file_client_args),
+    dict(type='mmdet.Resize', scale=(800, 800), keep_ratio=True),
+    dict(
+        type='mmdet.PackDetInputs',
+        meta_keys=('img_id', 'img_path', 'ori_shape', 'img_shape',
+                   'scale_factor'))
+]
+train_dataloader = dict(
+    batch_size=2,
+    num_workers=2,
+    persistent_workers=True,
+    sampler=dict(type='DefaultSampler', shuffle=True),
+    batch_sampler=None,
+    dataset=dict(
+        type='ConcatDataset',
+        ignore_keys=['DATASET_TYPE'],
+        datasets=[
+            dict(
+                type=dataset_type,
+                data_root=data_root,
+                ann_file='ImageSets/Main/train.txt',
+                data_prefix=dict(img_path='JPEGImages-trainval'),
+                filter_cfg=dict(filter_empty_gt=True),
+                pipeline=train_pipeline),
+            dict(
+                type=dataset_type,
+                data_root=data_root,
+                ann_file='ImageSets/Main/val.txt',
+                data_prefix=dict(img_path='JPEGImages-trainval'),
+                filter_cfg=dict(filter_empty_gt=True),
+                pipeline=train_pipeline)
+        ]))
```

Comment:
I think the `ConcatDataset` could be simplified to 
```
dataset=dict(
    type=dataset_type,
    data_root=data_root,
    ann_file=['ImageSets/Main/train.txt', 'ImageSets/Main/val.txt'],
    data_prefix=dict(img_path='JPEGImages-trainval'),
    filter_cfg=dict(filter_empty_gt=True),
    pipeline=train_pipeline),
```

Reply:
I've tried this before, but  it no longer seems to be supported in mmengine.

Reply:
> I think the `ConcatDataset` could be simplified to
> 
> ```
> dataset=dict(
>     type=dataset_type,
>     data_root=data_root,
>     ann_file=['ImageSets/Main/train.txt', 'ImageSets/Main/val.txt'],
>     data_prefix=dict(img_path='JPEGImages-trainval'),
>     filter_cfg=dict(filter_empty_gt=True),
>     pipeline=train_pipeline),
> ```

I've tried this before, but it doesn't seem to be supported in `mmengine`.